compat/src/suspense.js

```diff
@@ -43,12 +43,11 @@ options.unmount = function(vnode) {
 
 function detachedClone(vnode, detachedParent, parentDom) {
 	if (vnode) {
-		if (vnode._component && vnode._component.__hooks) {
-			vnode._component.__hooks._list.forEach(effect => {
-				if (typeof effect._cleanup == 'function') effect._cleanup();
-			});
-
+		if (vnode._component && vnode._component.__hooks && !vnode._dom) {
 			vnode._component.__hooks = null;
+		} else if (vnode._component && vnode._component.__hooks) {
```

Comment:
If this is a previously mounted component suspending due to an update we just clear potentially scheduled effects

Reply:
I think only the component that suspended has effects cleared, so this logic shouldn't run as part of the traversal. At least that is my observation here: https://codesandbox.io/s/parent-effects-when-child-suspends-1lsex

Reply:
Good point